recipes/sdl_net/all/conanfile.py

```diff
@@ -0,0 +1,86 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class SdlnetConan(ConanFile):
+    name = "sdl_net"
+    description = "A networking library for SDL"
+    license = "Zlib"
+    topics = ("sdl2", "sdl2_net", "sdl", "sdl_net", "net", "networking")
+    homepage = "https://www.libsdl.org/projects/SDL_net"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        self.requires("sdl/2.0.16")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.configure(build_folder=self._build_subfolder)
+        
+        return self._cmake
+
+    def build(self):
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
+            raise ConanInvalidConfiguration("sdl_net is not supported with Visual Studio")
+        # FIXME: check that major version of sdl_net is the same than sdl (not possible yet in validate())
```

Comment:
This FIXME + check should be kept in `build`.
It is very unlikely that it will cause the conan hooks to complain.

Reply:
```suggestion
        # FIXME: check that major version of sdl_net is the same than sdl (not possible yet in validate())
        if tools.Version(self.deps_cpp_info["sdl"]).major != tools.Version(self.version):
            raise ConanInvalidConfiguration(f"The major versions of {self.name} and sdl must be the same")
```